src/components/SettingsPage/RepairDeviceButton.js

```diff
@@ -45,7 +51,7 @@ class RepairDeviceButton extends PureComponent<Props, State> {
   repair = (version = null) => {
     if (this.state.isLoading) return
     const { push } = this.props
-    this.setState({ isLoading: true })
+    this.timeout = setTimeout(() => this.setState({ isLoading: true }), 500)
```

Comment:
it's a very edge case scenario but you should also clearTimeout in the complete case

Reply:
as well as in close() case.

Reply:
ok i'll do that